### ISSUE TITLE
Support arithmetical operations

### ISSUE THREAD

**Issue by AngryMaciek** [AngryMaciek/hypercomplex#11]
**Is your feature request related to a problem? Please describe.**
Need support for basic operations: `+,-,+=,-=,*,/,*=,/=,^,^=`.

**Describe the solution you'd like**
Overload respective operators.

**Additional context**
* https://www.learncpp.com/cpp-tutorial/92-overloading-the-arithmetic-operators-using-friend-functions/
* https://www.learncpp.com/cpp-tutorial/9-2a-overloading-operators-using-normal-functions/
* https://www.learncpp.com/cpp-tutorial/94-overloading-operators-using-member-functions/

Additionally, after reading these, please double check comparison operators, according to:
https://www.learncpp.com/cpp-tutorial/96-overloading-the-comparison-operators/

**Comment by AngryMaciek** [comment]
The following rules of thumb can help you determine which form is best for a given situation:

* If you’re overloading assignment (=), subscript ([]), function call (()), or member selection (->), do so as a member function.
* If you’re overloading a unary operator, do so as a member function.
* If you’re overloading a binary operator that does not modify its left operand (e.g. operator+), do so as a normal function (preferred) or friend function.
* If you’re overloading a binary operator that modifies its left operand, but you can’t modify the definition of the left operand (e.g. operator<<, which has a left operand of type ostream), do so as a normal function (preferred) or friend function.
* If you’re overloading a binary operator that modifies its left operand (e.g. operator+=), and you can modify the definition of the left operand, do so as a member function.

**Comment by AngryMaciek** [comment]
Equality and inequality operators fixed in 4910652.

**Comment by AngryMaciek** [comment]
`+` and `-` finished in 169b0cb

**Comment by AngryMaciek** [comment]
`+=` and `-=` finished in 45c9885

**Comment by AngryMaciek** [comment]
**TODO: throw esception if dimensions of operands do not match.**

**Comment by AngryMaciek** [comment]
`*` and `*=` finished in 214ac69

**Multiplication is not implemented very efficiently - requires refactoring.**

**Comment by AngryMaciek** [comment]
`^` and `^=` finished in 5c38917

Remember that this operator is defined only for **positive** integers.
2nd arg. can be `unsigned int` but also `unsigned short`, a positive `int` etc. -> **rewrite with templating?**

**Comment by AngryMaciek** [comment]
The last operators to overload are: `/` and `/=`.
Should be implemented as multiplication with an inverse, based on [octonions wiki](https://en.wikipedia.org/wiki/Octonion#Conjugate,_norm,_and_inverse)?
Require: inverse method; `inv` method requires `norm` method.

**Comment by AngryMaciek** [comment]
`/` and `/=` are finished in be67970, together with `inv` and `norm` methods.

**Comment by AngryMaciek** [comment]
Regarding https://github.com/AngryMaciek/hypercomplex/issues/11#issuecomment-719750153: implicit type conversion mechanism solves the issue, the following types are acceptable and tested:
```
        short int si = 2;
        unsigned short int usi = 2;
        int i = 2;
        unsigned int ui = 2;
        long int li = 2;
        unsigned long int uli = 2;
        long long int lli = 2;
        unsigned long long int ulli = 2;
```

**Comment by AngryMaciek** [comment]
https://github.com/AngryMaciek/hypercomplex/issues/11#issuecomment-719582785 fixed in ad9c2a5

**Comment by AngryMaciek** [comment]
`expand` member function added in 1228525 to aid arithmetical operations on distinct class instances.
